crossbeam-queue/src/array_queue.rs

```diff
@@ -200,6 +191,67 @@ impl<T> ArrayQueue<T> {
         }
     }
 
+    /// Attempts to push an element into the queue.
+    ///
+    /// If the queue is full, the element is returned back as an error.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use crossbeam_queue::ArrayQueue;
+    ///
+    /// let q = ArrayQueue::new(1);
+    ///
+    /// assert_eq!(q.push(10), Ok(()));
+    /// assert_eq!(q.push(20), Err(20));
+    /// ```
+    pub fn push(&self, value: T) -> Result<(), T> {
+        self.push_or_else(value, |v, _, _, _, _| Err(v))
+    }
+
+    /// Pushes an element into the queue or swaps the oldest element.
+    ///
+    /// If the queue is full, the oldest element is swapped and returned,
+    /// otherwise `None` is returned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use crossbeam_queue::ArrayQueue;
+    ///
+    /// let q = ArrayQueue::new(1);
+    ///
+    /// assert_eq!(q.push_or_swap(10), None);
+    /// assert_eq!(q.push_or_swap(20), Some(10));
+    /// assert_eq!(q.pop(), Some(20));
```

Comment:
Having an example with two capacity may make it easier to understand the behavior.

Reply:
yup :+1: